samples: subsys: usb: mass: Include rpi_pico.overlay

mcjelcom opened the pull request:
Include rpi_pico.overlay from the rpi_pico_rp2040_w.overlay in order to have `pico/rp2040/w` board to work out of the box.

nordicjm commented on samples/subsys/usb/mass/boards/rpi_pico_rp2040_w.overlay:
```diff
@@ -0,0 +1,19 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/delete-node/ &code_partition;
+
+#include <raspberrypi/partitions_2M_storage.dtsi>
```
the base board should be using these by default, same as the comment as the pr that was posted to mcuboot

jlbwoods replied:
Thanks for catching that!  I forgot to remove those lines after making storage partition default in boards.

For the updated PR, I also included the necessary `rpi_pico_rp2040_w.conf` file in order to setup FLASH and FILE_SYSTEM correctly.  Please take a look again!  Thank you!

jlbwoods commented on samples/subsys/usb/mass/boards/rpi_pico_rp2040_w.conf:
Sorry, I can only submit one file per commit?

jfischer-no replied:
I did not say that. Unjustified and independent board changes are mixed in with unnecessary sample changes in the same commit.

jlbwoods commented on samples/subsys/usb/mass/boards/rpi_pico_rp2040_w.conf:
```diff
@@ -0,0 +1,6 @@
+CONFIG_DISK_DRIVER_FLASH=y
+
+CONFIG_FILE_SYSTEM=y
+CONFIG_FAT_FILESYSTEM_ELM=y
```
Okay, got it.  Will remove this file.

jlbwoods replied:
There are two reasons to add overlays for this board:
 - Without `msc_disk0 {}` defined, the example using FAT FS won't build correctly:
   - `west build -b rpi_pico/rp2040/w samples/subsys/usb/mass -- -DCONFIG_APP_MSC_STORAGE_FLASH_FATFS=y`
 - The example also demonstrates how to switch to a different partitions other than the one defined in the boards.

To make things simple for this PR, we could decide to include just the `msc_disk0 {}` defines in order for the example to work correctly.

Please note that this overlays is just trying to mimic what other overlays in the same folder like `adafruit_feather_nrf52840_sense.overlay`.

Also, this overlays is trying to refactor existing `rpi_pico.overlay` using the now available partitions defined in DTS.  So, should we also modify `rpi_pico.overlay` with another PR?

jfischer-no replied:
> There are two reasons to add overlays for this board:
> 
>     * Without `msc_disk0 {}` defined, the example using FAT FS won't build correctly:

FATFS on RAMdisk should work just fine. There is nothing like "all the options of the sample must work with any board out of the box".
      
>       * `west build -b rpi_pico/rp2040/w samples/subsys/usb/mass -- -DCONFIG_APP_MSC_STORAGE_FLASH_FATFS=y`
> 
>     * The example also demonstrates how to switch to a different partitions other than the one defined in the boards.
> 

As I said there are enough examples how to do it.

> To make things simple for this PR, we could decide to include just the `msc_disk0 {}` defines in order for the example to work correctly.
> 
> Please note that this overlays is just trying to mimic what other overlays in the same folder like `adafruit_feather_nrf52840_sense.overlay`.
> 
> Also, this overlays is trying to refactor existing `rpi_pico.overlay` using the now available partitions defined in DTS. So, should we also modify `rpi_pico.overlay` with another PR?

I cannot find any commit touching rpi_pico.overlay. But look there is an example how to use it on pico based boards. So no need for any actions.

Commit:
samples: subsys: usb: mass: Include rpi_pico.overlay

Include rpi_pico.overlay from the rpi_pico_rp2040_w.overlay in order to
have `pico/rp2040/w` board to work out of the box.

Signed-off-by: John Lin <john.lin@beechwoods.com>
